### PR TITLE
Fix display video duration cause app crash when targeting to sdk 29 and fix display media file under root in folder mode

### PR DESCRIPTION
**Pull request by wiwuwa**
Fix Android Q cannot open gallery when the directory contains video

### CODE REVIEW

**Comment by wiwuwa** [comment]
Fix cannot display media file under root in folder mode
https://github.com/esafirm/android-image-picker/issues/333#issue-815025285

**Comment by bipinvaylu** [comment]
@esafirm I am also facing related issue in latest version `2.4.2` version as well. For now, I have to fix this issue by using this open PR code. It would be nice if you merge this PR or just notify once this PR target issue fixed. So i shall move back to dependency instead of using as sub-module (with this PR code).

cc: @hjchin  @sr1dh4r @xpdlf1004 @LinX64

**Comment by rahul4452** [comment]
@wiwuwa still facing the issue on Redmi note 7 pro after using this PR.

File: imagepicker/src/main/java/com/esafirm/imagepicker/features/fileloader/DefaultImageFileLoader.java [hunk 1]
```diff
@@ -166,6 +166,15 @@ public void run() {
                                 folderMap.put(bucket, folder);
                             }
                             folder.getImages().add(image);
+                        }else if(bucket == null){
```
Comment (esafirm) [inline imagepicker/src/main/java/com/esafirm/imagepicker/features/fileloader/DefaultImageFileLoader.java:169]: I think we can simplify this and further null-proof the code by changing this to

```java

// this code should be placed before if (name != null)

if (bucket == null) {
  File parent = new File(path).getParentFile();
  if (parent != null) {
    bucket = parent.getName();
  } else {
    bucket = "SDCARD";
  }
}
```

**Review by esafirm — CHANGES_REQUESTED** [review]

File: imagepicker/src/main/java/com/esafirm/imagepicker/features/fileloader/DefaultImageFileLoader.java (hunk 1 above)
Comment (wiwuwa) [reply imagepicker/src/main/java/com/esafirm/imagepicker/features/fileloader/DefaultImageFileLoader.java]: @esafirm Thanks for your effort, I have push it to this PR

**Comment by esafirm** [comment]
Thanks @wiwuwa LGTM 👍

**Comment by siva-komaragiri** [comment]
This issue still happened with 2.4.5, retriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION) is getting null which resulted in NumberFormatException with Long.parseLong

Running on Android 10 with app targeting SDK version 30

**Comment by esafirm** [comment]
@siva-komaragiri will add the fix to version 3.0.0

**Pull request merged**